wolfcrypt/src/curve25519.c

```diff
@@ -51,12 +51,43 @@ const curve25519_set_type curve25519_sets[] = {
     }
 };
 
+static const unsigned char kCurve25519BasePoint[CURVE25519_KEYSIZE] = {9};
+
+/* compute the public key from an existing private key, using bare vectors. */
+int wc_curve25519(int public_size, byte* public, int private_size, const byte* private) {
+    int ret;
+
+    if ((public_size != CURVE25519_KEYSIZE) ||
+        (private_size != CURVE25519_KEYSIZE)) {
+        return ECC_BAD_ARG_E;
+    }
+    if ((public == NULL) || (private == NULL))
+        return ECC_BAD_ARG_E;
+
+    /* check clamping */
+    if ((private[0] & ~248) ||
+        (private[CURVE25519_KEYSIZE-1] & 128))
```

Comment:
The check on clamp here does not quite match up with the clamp we use when making the key. Clamped at (x & 63)  (x | 64).

Example being (2048 & 128) = 0 but when making the private key a value of 2048 would be invalid.

Reply:
Disregard, it is a byte type so my thoughts here are incorrect. The check is good.

Reply:
as noted in side channel, clamping check seems to be correct as-is.